Dockerfile

```diff
@@ -1,24 +1,30 @@
-# First stage
+# First stage: Build the binary
 FROM golang:1.24 AS build-root
 
 WORKDIR /build
 
-COPY go.mod .
-COPY go.sum .
+# Copy dependency files first for better layer caching
+COPY go.mod go.sum ./
 
 RUN go mod download
 
+# Copy source code
 COPY . .
 
+# Build static binary with optimizations
 ENV CGO_ENABLED=0
 ENV GOOS=linux
 ENV GOARCH=amd64
 
-RUN go build ./...
+RUN go build -o vice-proxy -ldflags="-w -s" .
 
-## Second stage
-FROM golang:1.24
+## Second stage: Minimal runtime image
+FROM alpine:3.20
```

Comment:
I've encountered some SSL problems when copying certificates over from the golang image, which is based on Debian, to Alpine images. If you run into problems with that, switching to debian:stable-slim usually works.